providers/base/bin/watchdog_test.py

```diff
@@ -0,0 +1,493 @@
+                    return
+        except subprocess.CalledProcessError as err:
+            print(err, file=sys.stderr)
+        raise SystemExit("Error: No expected hardware watchdog been detected")
+    else:
+        if collect_hardware_watchdogs(watchdog_identity):
+            print("\nPassed: hardware watchdog detected")
+            return
+        raise SystemExit("Error: No hardware watchdog available")
+
+
+def watchdog_reset_test(
+    kernel_module, watchdog_identity, log_dir, watchdog_timeout
+) -> None:
+    """
+    perform a watchdog reset test
+    Note: if no kernel panic, this scripts would trigger a system reset using systemctl command
+
+    Args:
+        kernel_module: watchdog kernel module to be probe
+        watchdog_identity: expected identity string of the watchdog device
+        log_dir: the directory where watchdog configuration and log file will be saved
+        watchdog_timeout: the timeout value for the watchdog
+    """  # noqa: E501
+    print("# Perform watchdog reset test")
+    try:
+        watchdog_dev = collect_hardware_watchdogs(watchdog_identity)
+        if watchdog_dev:
+            trigger_system_reset(
+                log_dir, next(iter(watchdog_dev)), watchdog_timeout
+            )
+        else:
+            with probe_watchdog_module(kernel_module):
+                watchdog_dev = collect_hardware_watchdogs(watchdog_identity)
+                if watchdog_dev:
+                    trigger_system_reset(
+                        log_dir, next(iter(watchdog_dev)), watchdog_timeout
+                    )
+    except subprocess.CalledProcessError as err:
+        print(err, file=sys.stderr)
+    except Exception as err:
+        # bypass all exceptions as this is a no return job in checkbox
+        print(err, file=sys.stderr)
+    finally:
+        # Reboot system manually due to the system is not restart by watchdog
+        # When this file exists, it means test failed
+        time.sleep(60)
+        log_file = Path(log_dir).joinpath("watchdog_manual_reset.log")
+        log_file.write_text("system reset by watchdog")
+        subprocess.run(["sync"])
+        subprocess.run(["systemctl", "reboot"])
+
+
+def post_check_test(log_dir):
+    """
+    Restore the watchdog settings and
+    verify from logs that no reset occurred due to scripts or user activity.
+
+    Args:
+        log_dir: directory to store watchdog config and log file
+
+    Raises:
+        SystemExit: system been reset due to scripts or user activity.
+    """
+    print("# Post check test")
+    _get_watchdog_handler().restore_config(log_dir)
+
+    if Path(log_dir).joinpath("watchdog_manual_reset.log").exists():
+        raise SystemExit("Error: System reset by scripts, not watchdog")
+
+    start_time = Path(log_dir).joinpath(WATCHDOG_LOG_FILE).read_text()
+    system_uptime = float(Path("/proc/uptime").read_text().split(" ")[0])
+    time_diff = time.time() - system_uptime - float(start_time)
+    if time_diff > 300:
```

Comment:
I'm not a fan of hardcoded values like this, especially to guess if it was caused by user interaction. Slower systems may trigger this even though everything went fine.

My take would be to introduce a variable that can be set when running the script (this can be done using an environment variable); this way, on slow systems, this can be adjusted.